File: src/main/java/me/wonwoo/web/IndexController.java
package me.wonwoo.web;

import me.wonwoo.domain.PostRepository;
import me.wonwoo.utils.MarkDownConverter;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.reactive.result.view.Rendering;

@Controller
public class IndexController {

  private final PostRepository postRepository;
  private final MarkDownConverter markDownConverter;

  public IndexController(PostRepository postRepository, MarkDownConverter markDownConverter) {
    this.postRepository = postRepository;
    this.markDownConverter = markDownConverter;
  }

  @GetMapping("/")
  public Rendering home() {
    return Rendering
        .view("index")
        .modelAttribute("posts", this.postRepository.findAll()
            .map(post -> new PostDto(post, markDownConverter)))
        .build();
  }
}

File: src/main/kotlin/me/wonwoo/web/IndexController.kt
package me.wonwoo.web

import me.wonwoo.domain.PostRepository
import me.wonwoo.utils.MarkDownConverter
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.reactive.result.view.Rendering

@Controller
class IndexController(private val postRepository: PostRepository,
                      private val markDownConverter: MarkDownConverter) {

  @GetMapping("/")
  fun home(): Rendering {
    return Rendering
        .view("index")
        .modelAttribute("posts", this.postRepository.findAll()
            .map { it.toDto(markDownConverter) })
        .build()
  }
}